src/main/webapp/app/shared/util/core-submission/core-submission.spec.ts

```diff
@@ -0,0 +1,659 @@
+import {
+  CancerType,
+  DX_LEVELS,
+  Drug,
+  DrugCollection,
+  FDA_LEVELS,
+  FIREBASE_ONCOGENICITY,
+  Gene,
+  HistoryOperationType,
+  Implication,
+  Mutation,
+  MutationEffect,
+  PX_LEVELS,
+  Review,
+  TI,
+  TI_TYPE,
+  TX_LEVELS,
+  Treatment,
+  Tumor,
+} from 'app/shared/model/firebase/firebase.model';
+import { getEvidence, pathToGetEvidenceArgs } from './core-submission';
+import { EvidenceEvidenceTypeEnum, EvidenceLevelOfEvidenceEnum, TumorType } from 'app/shared/api/generated/core';
+import { generateUuid } from '../utils';
+
+type GetEvidenceArgs = Parameters<typeof getEvidence>[0];
+type GetEvidenceRtn = ReturnType<typeof getEvidence>;
+describe('getEvidence to submit to core', () => {
+  describe('Path parse tests', () => {
+    const { mutation, updateTime, gene, tumor, tiIs, tiIr, tiSs, tiSr, drugListRef, treatment, entrezGeneId } = createMockGene();
+
+    const goodTests: [valuePath: string, args: Partial<GetEvidenceArgs>][] = [
+      ['mutations/0', undefined],
+      ['summary', { type: EvidenceEvidenceTypeEnum.GeneSummary, updateTime, gene, drugListRef, entrezGeneId }],
+      ['background', { type: EvidenceEvidenceTypeEnum.GeneBackground, updateTime, gene, drugListRef, entrezGeneId }],
+      [
+        'mutations/0/mutation_effect/oncogenic',
+        { type: EvidenceEvidenceTypeEnum.Oncogenic, mutation, updateTime, gene, drugListRef, entrezGeneId },
+      ],
+      [
+        'mutations/0/tumors/0/summary',
+        { type: EvidenceEvidenceTypeEnum.GeneTumorTypeSummary, tumor, mutation, updateTime, gene, drugListRef, entrezGeneId },
+      ],
+      [
+        'mutations/0/tumors/0/prognosticSummary',
+        { type: EvidenceEvidenceTypeEnum.PrognosticSummary, tumor, mutation, updateTime, gene, drugListRef, entrezGeneId },
+      ],
+      [
+        'mutations/0/tumors/0/diagnosticSummary',
+        { type: EvidenceEvidenceTypeEnum.DiagnosticSummary, tumor, mutation, updateTime, gene, drugListRef, entrezGeneId },
+      ],
+      [
+        'mutations/0/mutation_effect/oncogenic',
+        { type: EvidenceEvidenceTypeEnum.Oncogenic, mutation, updateTime, gene, drugListRef, entrezGeneId },
+      ],
+      [
+        'mutations/0/mutation_effect/effect',
```

Comment:
`mutations/0/mutation_effect/description` should trigger a EvidenceEvidenceTypeEnum.MutationEffect change as well